interface/reports/reports_controllers/ApptEncounterController.php

```diff
@@ -0,0 +1,316 @@
+                      if ($brow['modifier']) {
+                        $query .= "modifier = '" . $brow['modifier'] . "'";
+                      } else {
+                        $query .= "(modifier IS NULL OR modifier = '')";
+                      }
+                      $query .= " LIMIT 1";
+                      $tmp = sqlQuery($query);
+                      $relcodes = explode(';', $tmp['related_code']);
+                      foreach ($relcodes as $codestring) {
+                        if ($codestring === '') continue;
+                        list($codetype, $code) = explode(':', $codestring);
+                        if ($codetype !== 'IPPF') continue;
+                        if (preg_match('/^25222/', $code)) $gcac_related_visit = true;
+                      }
+                    }
+                  } // End IPPF stuff
+                } // end while
+               
+                $copays -= getPatientCopay($patient_id,$encounter);
+
+                // The following is removed, perhaps temporarily, because gcac reporting
+                // no longer depends on gcac issues.  -- Rod 2009-08-11
+                /******************************************************************
+                // More custom code for IPPF.  Generates an error message if a
+                // GCAC issue is required but is not linked to this visit.
+                if (!$errmsg && $gcac_related_visit) {
+                  $grow = sqlQuery("SELECT l.id, l.title, l.begdate, ie.pid " .
+                    "FROM lists AS l " .
+                    "LEFT JOIN issue_encounter AS ie ON ie.pid = l.pid AND " .
+                    "ie.encounter = '$encounter' AND ie.list_id = l.id " .
+                    "WHERE l.pid = '$patient_id' AND " .
+                    "l.activity = 1 AND l.type = 'ippf_gcac' " .
+                    "ORDER BY ie.pid DESC, l.begdate DESC LIMIT 1");
+                  // Note that reverse-ordering by ie.pid is a trick for sorting
+                  // issues linked to the encounter (non-null values) first.
+                  if (empty($grow['pid'])) { // if there is no linked GCAC issue
+                    if (empty($grow)) { // no GCAC issue exists
+                      $errmsg = "GCAC issue does not exist";
+                    } else { // there is one but none is linked
+                      $errmsg = "GCAC issue is not linked";
+                    }
+                  }
+                }
+                ******************************************************************/
+                if ($gcac_related_visit) {
+                  $grow = sqlQuery("SELECT COUNT(*) AS count FROM forms " .
+                    "WHERE pid = '$patient_id' AND encounter = '$encounter' AND " .
+                    "deleted = 0 AND formdir = 'LBFgcac'");
+                  if (empty($grow['count'])) { // if there is no gcac form
+                    postError(xl('GCAC visit form is missing'));
```

Comment:
What is GCAC?

Reply:
This already existed. Maybe @teryhill can answer this